cpp/src/interop/from_arrow_host.cu

```diff
@@ -188,8 +188,8 @@ std::unique_ptr<column> dispatch_copy_from_arrow_host::operator()<cudf::string_v
 
   // chars_column does not contain any nulls, they are tracked by the parent string column
   // itself instead. So we pass nullptr for the validity bitmask.
-  size_type const char_data_length =
-    reinterpret_cast<int32_t const*>(offset_buffers[1])[input->length + input->offset];
+  int64_t const char_data_length =
+    reinterpret_cast<int64_t const*>(offset_buffers[1])[input->length + input->offset];
```

Comment:
This looks wrong to me. Casting to int64 should only be for LARGE_STRINGS type right?

Reply:
I think you're looking at an outdated version of the code, but yes that was incorrect. It now casts based on the type.

Reply:
Ah yes. Sorry.